File: DarkOrbit 10.0/Net/netty/handlers/QualitySettingsRequestHandler.cs
using Ow.Game;
using Ow.Managers;
using Ow.Net.netty.commands;
using Ow.Net.netty.requests;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Ow.Net.netty.handlers
{
    class QualitySettingsRequestHandler : IHandler
    {
        public void execute(GameSession gameSession, byte[] bytes)
        {
            var read = new QualitySettingsRequest();
            read.readCommand(bytes);

            var player = gameSession.Player;
            var qualitySettings = player.Settings.Quality;

            qualitySettings.notSet = false;
            qualitySettings.qualityAttack = read.qualityAttack;
            qualitySettings.qualityBackground = read.qualityBackground;
            qualitySettings.qualityCollectable = read.qualityCollectable;
            qualitySettings.qualityCustomized = read.qualityCustomized;
            qualitySettings.qualityEffect = read.qualityEffect;
            qualitySettings.qualityEngine = read.qualityEngine;
            qualitySettings.qualityExplosion = read.qualityExplosion;
            qualitySettings.qualityPoizone = read.qualityPoizone;
            qualitySettings.qualityPresetting = read.qualityPresetting;
            qualitySettings.qualityShip = read.qualityShip;

            QueryManager.SavePlayer.Settings(player, "quality", qualitySettings);
        }
    }
}

File: DarkOrbit 9.0/DarkOrbit 9.0 Emulator/src/simulator/netty/handlers/QualitySettingsRequestHandler.java
package simulator.netty.handlers;

import mysql.QueryManager;
import net.game_server.GameServerClientConnection;
import net.game_server.GameSession;
import simulator.netty.ClientCommand;
import simulator.netty.clientCommands.QualitySettingsRequest;
import simulator.users.ClientSettings;
import utils.Log;

/**
 Created by Pedro on 31-03-2015.
 */
public class QualitySettingsRequestHandler
        implements ICommandHandler {

    private final GameServerClientConnection mGameServerClientConnection;
    private final QualitySettingsRequest     mQualitySettingsRequest;

    public QualitySettingsRequestHandler(final GameServerClientConnection pGameServerClientConnection,
                                         final ClientCommand pQualitySettingsRequest) {
        this.mGameServerClientConnection = pGameServerClientConnection;
        this.mQualitySettingsRequest = (QualitySettingsRequest) pQualitySettingsRequest;
    }

    @Override
    public void execute() {
        Log.pt("QUALITY SETTINGS REQUEST HANDLER");
        final GameSession gameSession = this.mGameServerClientConnection.getGameSession();
        if (gameSession != null) {
            final ClientSettings.Quality quality = gameSession.getAccount()
                                                              .getClientSettingsManager()
                                                              .getClientSettings()
                                                              .getQuality();
            quality.setNotSet(false);
            quality.setQualityEngine(this.mQualitySettingsRequest.qualityEngine);
            quality.setQualityEffect(this.mQualitySettingsRequest.qualityEffect);
            quality.setQualityCustomized(this.mQualitySettingsRequest.qualityCustomized);
            quality.setQualityCollectable(this.mQualitySettingsRequest.qualityCollectable);
            quality.setQualityPoizone(this.mQualitySettingsRequest.qualityPoizone);
            quality.setQualityPresetting(this.mQualitySettingsRequest.qualityPresetting);
            quality.setQualityBackground(this.mQualitySettingsRequest.qualityBackground);
            quality.setQualityAttack(this.mQualitySettingsRequest.qualityAttack);
            quality.setQualityExplosion(this.mQualitySettingsRequest.qualityExplosion);
            quality.setQualityShip(this.mQualitySettingsRequest.qualityShip);
            QueryManager.saveAccount(gameSession.getAccount());
        }
    }
}
